libs/language-server/src/lib/ast/model-util.ts

```diff
@@ -4,26 +4,47 @@
 
 import { strict as assert } from 'assert';
 
-import { AstNode, assertUnreachable } from 'langium';
+import { AstNode, Reference, assertUnreachable } from 'langium';
 
 // eslint-disable-next-line import/no-cycle
 import { getMetaInformation } from '../meta-information/meta-inf-registry';
 
 import {
   BinaryExpression,
   BlockDefinition,
+  CompositeBlocktypeDefinition,
   PipelineDefinition,
   UnaryExpression,
+  isCompositeBlocktypeDefinition,
 } from './generated/ast';
 import { PipeWrapper, createSemanticPipes } from './wrappers/pipe-wrapper';
 
 export function collectStartingBlocks(
-  pipeline: PipelineDefinition,
+  container: PipelineDefinition | CompositeBlocktypeDefinition,
 ): BlockDefinition[] {
+  // For composite blocks the first blocks of all pipelines are starting blocks as they have inputs
+  if (isCompositeBlocktypeDefinition(container)) {
+    const startingBlocks = container.pipes
+      .map((pipe) => pipe.blocks[0])
+      .map((blockRef: Reference<BlockDefinition> | undefined) => {
+        if (
+          blockRef &&
+          blockRef.ref &&
+          getMetaInformation(blockRef.ref.type) !== undefined
```

Comment:
Can be simplified by using the `?` operator. And I'd be explicit with the `!== undefined` check (just a nitpick).

```suggestion
          blockRef?.ref !== undefined &&
          getMetaInformation(blockRef.ref.type) !== undefined
```